### PR TITLE
OPENPGP PLUGIN: Show key as expired or revoked in account manager window

### PR DESCRIPTION
**Pull request by eerielili**
A small update concerning the openpgp plugin: with this, the key can be shown either as expired or revoked in the account manager windows concerning an account.
image

### CODE REVIEW

**Comment by mxlgv** [comment]
@giantplaceholder do we need this change?

File: plugins/openpgp/src/account_settings_entry.vala [hunk 1]
```diff
@@ -69,7 +69,7 @@ public class AccountSettingsEntry : Plugins.AccountSettingsEntry {
             return;
         }
         if (keys.size == 0) {
-            label.set_markup(build_markup_string(_("Key publishing disabled"), _("No keys available. Generate one!")));
+            label.set_markup(build_markup_string(_("Key publishing disabled"), _("No keys available. Generate one or check if your keys aren't expired or revoked!")));
```
Comment (mxlgv) [inline plugins/openpgp/src/account_settings_entry.vala:72]: You also need to fix the gettext files, at least for Russian, English and German.

**Review by mxlgv — CHANGES_REQUESTED** [review]

File: plugins/openpgp/src/account_settings_entry.vala (hunk 1 above)
Comment (eerielili) [reply plugins/openpgp/src/account_settings_entry.vala:72]: Okay. I'll update the dino.pot file and fr.po file at least, I don't know German and Russian that much.

**Comment by eerielili** [comment]
@giantplaceholder Thanks a lot for the review. I updated the translation template `plugins/openpgp/po/dino-openpgp.pot` just now, also added some translations for my language as well.

**Comment by mxlgv** [comment]
@eerielili please rebase CI for Windows has been fixed.

**Comment by eerielili** [comment]
@mxlgv done

File: plugins/openpgp/src/account_settings_entry.vala [hunk 2]
```diff
@@ -88,6 +88,18 @@ public class AccountSettingsEntry : Plugins.AccountSettingsEntry {
         set_label_active(selected);
 
         combobox.changed.connect(key_changed);
+        if (account_key != null){
```
Comment (mxlgv) [inline plugins/openpgp/src/account_settings_entry.vala:91]: missing backspace `) {`

File: plugins/openpgp/po/ru.po [hunk 3]
```diff
@@ -17,38 +18,62 @@ msgstr ""
...
+#: contact_details_provider.vala:28
 msgid "Key not in keychain"
 msgstr "Отсутствует ключ в связке"
 
-#: plugins/openpgp/src/contact_details_provider.vala:30
+#: contact_details_provider.vala:30
 msgid "Encryption"
 msgstr "Шифрование"
+
+#~ msgid "No keys available. Generate one!"
```
Comment (mxlgv) [inline plugins/openpgp/po/ru.po:75]: Remove completely if the code is not used

File: plugins/openpgp/po/fr.po [hunk 4]
```diff
@@ -5,49 +5,73 @@
 #
 msgid ""
 msgstr ""
-"Project-Id-Version: PACKAGE VERSION\n"
```
Comment (mxlgv) [inline plugins/openpgp/po/fr.po:8]: Why was `PACKAGE VERSION` removed?

File: plugins/openpgp/po/fr.po [hunk 5]
```diff
@@ -5,49 +5,73 @@
...
+#: contact_details_provider.vala:28
 msgid "Key not in keychain"
 msgstr "La clé n’est pas dans le trousseau"
 
-#: plugins/openpgp/src/contact_details_provider.vala:30
+#: contact_details_provider.vala:30
 msgid "Encryption"
 msgstr "Chiffrement"
+
+#~ msgid "No keys available. Generate one!"
```
Comment (mxlgv) [inline plugins/openpgp/po/fr.po:76]: Remove completely if the code is not used

File: plugins/openpgp/po/ru.po [hunk 6]
```diff
@@ -5,10 +5,11 @@
 #
 msgid ""
 msgstr ""
-"Project-Id-Version: PACKAGE VERSION\n"
```
Comment (mxlgv) [inline plugins/openpgp/po/ru.po:8]: Why was `PACKAGE VERSION` removed?

File: plugins/openpgp/po/ru.po [hunk 7]
```diff
@@ -17,38 +18,62 @@ msgstr ""
...
 msgstr "Ошибка в GnuPG"
 
-#: plugins/openpgp/src/account_settings_entry.vala:72
-msgid "No keys available. Generate one!"
-msgstr "Нет доступных ключей. Создайте как минимум один, либо проверьте что уже существующие (ранее созданные) ключи не были отозваны или срок их действия не закончился!"
+#: account_settings_entry.vala:72
+msgid ""
+"No keys available. Generate one or check if your keys aren't expired or "
+"revoked!"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/ru.po:36]: Please add this translation:
```
"Нет доступных ключей. Сгенерируйте их или проверьте срок действия."
"Также возможно, что ключ был отозван!"
```

File: plugins/openpgp/po/ru.po [hunk 8]
```diff
@@ -17,38 +18,62 @@ msgstr ""
...
+
+#: account_settings_entry.vala:95
+msgid "expired!"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/ru.po:40]: Please add this translation:
```
"истёк!"
```

File: plugins/openpgp/po/ru.po [hunk 9]
```diff
@@ -17,38 +18,62 @@ msgstr ""
...
+
+#: account_settings_entry.vala:95
+msgid "revoked!"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/ru.po:44]: Please add this translation:
```
"отозван!"
```

File: plugins/openpgp/po/ru.po [hunk 10]
```diff
@@ -17,38 +18,62 @@ msgstr ""
...
 
-#: plugins/openpgp/src/account_settings_entry.vala:101
+#: account_settings_entry.vala:96
+msgid "Attention required!"
+msgstr ""
+
+#: account_settings_entry.vala:96
+#, c-format
+msgid "Your key %s is %s"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/ru.po:53]: Please add this translation:
```
"Ваш ключ %s %s"
```

File: plugins/openpgp/po/de.po [hunk 11]
```diff
@@ -2,50 +2,72 @@ msgid ""
...
 msgstr "Fehler in GnuPG"
 
-#: plugins/openpgp/src/account_settings_entry.vala:72
-msgid "No keys available. Generate one!"
-msgstr "Keine Schlüssel vorhanden. Erzeuge einen!"
+#: account_settings_entry.vala:72
+msgid ""
+"No keys available. Generate one or check if your keys aren't expired or "
+"revoked!"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/de.po:30]: Please add this translation:
```
"Keine Schlüssel vorhanden. Erstellen Sie einen oder prüfen Sie,"
"ob Ihre Schlüssel nicht abgelaufen sind oder widerrufen!"
```

File: plugins/openpgp/po/de.po [hunk 12]
```diff
@@ -2,50 +2,72 @@ msgid ""
...
+
+#: account_settings_entry.vala:95
+msgid "expired!"
+msgstr ""
+
+#: account_settings_entry.vala:95
+msgid "revoked!"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/de.po:38]: Please add this translation:
```
"widerrufen!"
```

File: plugins/openpgp/po/de.po [hunk 13]
```diff
@@ -2,50 +2,72 @@ msgid ""
...
-msgstr "Keine Schlüssel vorhanden. Erzeuge einen!"
+#: account_settings_entry.vala:72
+msgid ""
+"No keys available. Generate one or check if your keys aren't expired or "
+"revoked!"
+msgstr ""
+
+#: account_settings_entry.vala:95
+msgid "expired!"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/de.po:34]: Please add this translation:
```
"abgelaufen!"
```

File: plugins/openpgp/po/de.po [hunk 14]
```diff
@@ -2,50 +2,72 @@ msgid ""
...
+#: contact_details_provider.vala:28
 msgid "Key not in keychain"
 msgstr "Schlüssel nicht im Schlüsselbund"
 
-#: plugins/openpgp/src/contact_details_provider.vala:30
+#: contact_details_provider.vala:30
 msgid "Encryption"
 msgstr "Verschlüsselung"
 
+#~ msgid "No keys available. Generate one!"
```
Comment (mxlgv) [inline plugins/openpgp/po/de.po:69]: Remove completely if the code is not used

File: plugins/openpgp/po/de.po [hunk 15]
```diff
@@ -2,50 +2,72 @@ msgid ""
...
 
-#: plugins/openpgp/src/account_settings_entry.vala:101
+#: account_settings_entry.vala:96
+msgid "Attention required!"
+msgstr ""
+
+#: account_settings_entry.vala:96
+#, c-format
+msgid "Your key %s is %s"
+msgstr ""
```
Comment (mxlgv) [inline plugins/openpgp/po/de.po:47]: Please add this translation:
```
"Ihr Schlüssel %s ist %s"
```

**Review by mxlgv — CHANGES_REQUESTED** [review]

**Review by mxlgv — APPROVED** [review]

**Comment by eerielili** [comment]
Thanks. Wait a minute, I must pust a russian translation i deleted by mistake

File: plugins/openpgp/po/ru.po (hunk 8 above)
Comment (mxlgv) [reply plugins/openpgp/po/ru.po:41]: ?

File: plugins/openpgp/po/ru.po (hunk 9 above)
Comment (mxlgv) [reply plugins/openpgp/po/ru.po:45]: ?

File: plugins/openpgp/po/ru.po (hunk 10 above)
Comment (mxlgv) [reply plugins/openpgp/po/ru.po:54]: ?

**Review by mxlgv — CHANGES_REQUESTED** [review]

**Comment by eerielili** [comment]
Updated the Russian translation. I used deepl.com to fill out the gaps since it's not my native language.

File: plugins/openpgp/po/ru.po (hunk 8 above)
Comment (mxlgv) [reply plugins/openpgp/po/ru.po]: @eerielili can you see this?  This is my language.  I offered a fairly accurate translation for each.

**Comment by mxlgv** [comment]
@eerielili  Thank you for your contribution to the project.     After the merger I will correct the translation.  Don't worry about translation)

**Review by eerielili — COMMENTED** [review]
done

**Review by mxlgv — APPROVED** [review]

**Comment by eerielili** [comment]
@mxlgv
I did update my branch to fit the new account settings UI.

**Pull request merged**